apis/identity/v1alpha1/iamaccesskey_types.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+)
+
+// IAMAccessKeyParameters define the desired state of an AWS IAM Access Key.
+type IAMAccessKeyParameters struct {
+	// IAMUsername contains the name of the IAMUser.
+	// +optional
+	IAMUsername string `json:"userName,omitempty"`
+
+	// IAMUsernameRef references to an IAMUser to retrieve its userName
+	// +optional
+	IAMUsernameRef *runtimev1alpha1.Reference `json:"userNameRef,omitempty"`
+
+	// IAMUsernameSelector selects a reference to an IAMUser to retrieve its userName
+	// +optional
+	IAMUsernameSelector *runtimev1alpha1.Selector `json:"userNameSelector,omitempty"`
+}
+
+// An IAMAccessKeySpec defines the desired state of an IAM Access Key.
+type IAMAccessKeySpec struct {
+	runtimev1alpha1.ResourceSpec `json:",inline"`
+	ForProvider                  IAMAccessKeyParameters `json:"forProvider"`
+}
+
+// IAMAccessKeyObservation keeps the state for the external resource
+type IAMAccessKeyObservation struct {
+	// The ID of the access key that unique identifies this resource
+	AccessKeyID string `json:"accessKeyId,omitempty"`
+
+	// The current status of this IAMAccessKey
+	// +kubebuilder:validation:Enum=Active;Inactive
```

Comment:
```suggestion
```

Since this is not entered by the user, I'd not put a validation there so that if AWS changes anything, we don't care.

Reply:
Well, it seems like `Status` is a configuration that user assigns as far as I can tell from `Update`. So, it should go to spec and we can have the validation there.